test/llvm-intrinsics/dynamic-memmove.ll

```diff
@@ -0,0 +1,51 @@
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc -spirv-text -o %t.txt
+; RUN: FileCheck < %t.txt %s --check-prefix=CHECK-SPIRV
+; RUN: llvm-spirv %t.bc -o %t.spv
+; RUN: llvm-spirv -r %t.spv -o %t.rev.bc
+; RUN: llvm-dis < %t.rev.bc | FileCheck %s --check-prefix=CHECK-LLVM
+
+; CHECK-SPIRV-NOT: llvm.memmove
+
+target datalayout = "e-m:e-i64:64-f80:128-n8:16:32:64-S128"
+target triple = "spir64-unknown-unknown"
+
+declare void @llvm.memmove.p1i8.p1i8.i64(i8 addrspace(1)* nocapture, i8 addrspace(1)* nocapture readonly, i64, i1)
+
+define spir_func void @memmove_caller(i8 addrspace(1)* %dst, i8 addrspace(1)* %src, i64 %n) {
+entry:
+  call void @llvm.memmove.p1i8.p1i8.i64(i8 addrspace(1)* %dst, i8 addrspace(1)* %src, i64 %n, i1 false)
+  ret void
+
+; CHECK-LLVM: @memmove_caller(i8 addrspace(1)* [[DST:%.*]], i8 addrspace(1)* [[SRC:%.*]], i64 [[N:%.*]])
+; CHECK-LLVM:  [[SRC_CMP:%.*]] = ptrtoint i8 addrspace(1)* %src to i64
```

Comment:
Is it worth checking whole loop code generated by LLVM in SPIR-V translator test?
Is it possible to left checks for only some minimal subset of instructions?

Reply:
Probably not, I didn't come up with any minimal subset though. Do you have any suggestions?

Reply:
I'd left only first 6 check strings. for example. It is enough to see that some loop preparation instructions are added that use src, dst and N from memmove instruction.

Reply:
Probably, we may even add only check for spirv code part and skip translation back into llvm.

Reply:
I prefer checking in LLVM IR, since it is more readable.